modules/api/php/endpoints/project/dicoms.class.inc

```diff
@@ -0,0 +1,153 @@
+ *
+ * PHP Version 7
+ *
+ * @category API
+ * @package  Loris
+ * @author   Simon Pelletier <simon.pelletier@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://github.com/aces/Loris
+ */
+namespace LORIS\api\Endpoints\Project;
+
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+use \LORIS\api\Endpoint;
+
+/**
+ * A class for handling the /projects/$projectname/dicoms endpoint.
+ *
+ * @category API
+ * @package  Loris
+ * @author   Simon Pelletier <simon.pelletier@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://github.com/aces/Loris
+ */
+class Dicoms extends Endpoint implements \LORIS\Middleware\ETagCalculator
+{
+    /**
+     * A cache of the results of the endpoint, so that
+     * it doesn't need to be recalculated for the ETag and handler
+     */
+    private $_cache;
+
+    /**
+     * The requested project
+     */
+    private $_project;
+
+    /**
+     * Contructor
+     *
+     * @param \Project $project The requested project
+     */
+    public function __construct(\Project $project)
+    {
+        $this->_project = $project;
+    }
+
+    /**
+     * Return which methods are supported by this endpoint.
+     *
+     * @return array supported HTTP methods
+     */
+    protected function allowedMethods() : array
+    {
+        return ['GET'];
+    }
+
+    /**
+     * Versions of the LORIS API which are supported by this
+     * endpoint.
+     *
+     * @return array a list of supported API versions.
+     */
+    protected function supportedVersions() : array
+    {
+        return ["v0.0.3"];
+    }
+
+    /**
+     * Handles a request that starts with /projects/$projectname/candidates
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        $pathparts = $request->getAttribute('pathparts');
+        if (count($pathparts) !== 0) {
+            return new \LORIS\Http\Response\JSON\NotFound();
+        }
+
+        switch ($request->getMethod()) {
+        case 'GET':
+            return $this->_handleGET($request);
+
+        case 'OPTIONS':
+            return (new \LORIS\Http\Response())
+                ->withHeader('Allow', $this->allowedMethods());
+        default:
+            return new \LORIS\Http\Response\JSON\MethodNotAllowed(
+                $this->allowedMethods()
+            );
+        }
+    }
+
+    /**
+     * Create an array representation of this endpoint's reponse body
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleGET(ServerRequestInterface $request): ResponseInterface
+    {
+        if (isset($this->_cache)) {
```

Comment:
Make sure that _hasAccess is called
The dicom_archives module requires `dicom_archive_view_allsites` to access it. 
@cmadjar The same applies here right?

Reply:
I added it, but I did like for the `project/recordings` [endpoint](https://github.com/aces/Loris/blob/main/modules/api/php/endpoints/project/recordings.class.inc#L97), which calls `_hasAccess` in `handle`. It would not matter here since I don't think the endpoints in `\projects` will ever have any other handles than handleGET, but I think it's better in `handle` for consistency